src/msg/Messenger.cc

```diff
@@ -20,12 +20,12 @@ Messenger *Messenger::create_client_messenger(CephContext *cct, std::string lnam
 
 uint64_t Messenger::get_pid_nonce()
 {
-  uint64_t nonce = getpid();
-  if (nonce == 1 || getenv("CEPH_USE_RANDOM_NONCE")) {
-    // we're running in a container; use a random number instead!
-    nonce = ceph::util::generate_random_number<uint64_t>();
-  }
-  return nonce;
+  // in the past the logic here was more complex -- we were trying
+  // to use the PID but, in the containerized world, it turned out
+  // unreliable. To deal with this, we started guessing whether we
+  // run in a container or not, and of course, got manual lever to
+  // intervene if guessed wrong (CEPH_USE_RANDOM_NONCE).
+  return ceph::util::generate_random_number<uint64_t>();
```

Comment:
Why not just nuke `get_pid_nonce`, replacing all calls with `get_random_nonce`?

Reply:
:+1:.